router/src/main/scala/calico/router/Router.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package calico.router
+
+import calico.std.History
+import cats.effect.kernel.Async
+import cats.effect.kernel.RefSink
+import cats.effect.kernel.Resource
+import cats.effect.kernel.Unique
+import cats.effect.syntax.all.*
+import cats.syntax.all.*
+import fs2.Stream
+import fs2.concurrent.Signal
+import org.http4s.Uri
+import org.scalajs.dom
+import fs2.concurrent.Topic
+
+trait Router[F[_]]:
+  def forward: F[Unit]
+  def back: F[Unit]
+  def go(delta: Int): F[Unit]
+  def navigate(uri: Uri): F[Unit]
+  def teleport(uri: Uri): F[Unit]
```

Comment:
What are the intended differences between these 2 method? 

Otherwise, they looks kind of like synonyms (as I'm currently reading it)